File: src/main/java/itmo/is/project/mapper/user/SpaceshipMapper.java
package itmo.is.project.mapper.user;

import itmo.is.project.dto.user.spaceship.CreateSpaceshipRequest;
import itmo.is.project.dto.user.spaceship.SpaceshipDto;
import itmo.is.project.mapper.EntityMapper;
import itmo.is.project.model.user.Spaceship;
import org.mapstruct.Mapper;

@Mapper(uses = {UserMapper.class})
public interface SpaceshipMapper extends EntityMapper<SpaceshipDto, Spaceship> {
    Spaceship toEntity(CreateSpaceshipRequest request);
}

File: src/main/kotlin/itmo/isproject/shared/user/mapper/SpaceshipMapper.kt
package itmo.isproject.shared.user.mapper

import itmo.isproject.shared.user.dto.spaceship.CreateSpaceshipRequest
import itmo.isproject.shared.user.dto.spaceship.SpaceshipDto
import itmo.isproject.shared.common.mapper.EntityMapper
import itmo.isproject.shared.user.model.Spaceship
import org.mapstruct.Mapper

@Mapper(uses = [UserMapper::class])
interface SpaceshipMapper : EntityMapper<SpaceshipDto, Spaceship> {

    fun toEntity(request: CreateSpaceshipRequest): Spaceship
}
